esp/services/ws_dfu/ws_dfuXRefService.cpp

```diff
@@ -195,16 +196,52 @@ IXRefFilesNode* CWsDfuXRefEx::getFileNodeInterface(IXRefNode& XRefNode,const cha
     return 0;
 }
 
+void CWsDfuXRefEx::readLostFileQueryResult(IEspContext &context, StringBuffer& buf)
+{
+    Owned<IPropertyTree> lostFilesQueryResult = createPTreeFromXMLString(buf.str());
+    if (!lostFilesQueryResult)
+    {
+        PROGLOG("readLostFileQueryResult() failed in creating PTree.");
+        return;
+    }
+
+    StringBuffer username;
+    Owned<IUserDescriptor> userdesc;
+    context.getUserID(username);
+    if(username.length() > 0)
+    {
+        const char* passwd = context.queryPassword();
+        userdesc.setown(createUserDescriptor());
+        userdesc->set(username.str(), passwd);
+    }
+
+    Owned<IPropertyTreeIterator> iter = lostFilesQueryResult->getElements("File");
+    ForEach(*iter)
+    {
+        IPropertyTree& item = iter->query();
+        const char* fileName = item.queryProp("Name");
+        if (!fileName || !*fileName)
+            continue;
+
+        Owned<IDistributedFile> df = queryDistributedFileDirectory().lookup(fileName, userdesc, false, false, true); // lock super-owners
```

Comment:
Am not sure why the lockSuperOwner = true is here?

Also, I suspect it should have a timeout of 0 (fail if locked) and capture the failure and continue, providing a warning that can't recover lost file as locked.
NB: it shouldn't be in use/locked, but it's better than stalling/failing here.